mlflow/utils/environment.py

```diff
@@ -82,3 +89,61 @@ def _get_additional_pip_dep(conda_env):
 def _log_pip_requirements(conda_env, path, requirements_file="requirements.txt"):
     pip_deps = _get_additional_pip_dep(conda_env)
     _mlflow_additional_pip_env(pip_deps, path=os.path.join(path, requirements_file))
+
+
+def _get_pip_requirements(module_names):
+    module_to_packages = importlib_metadata.packages_distributions()
+    requirements = []
+    for module_name in module_names:
+        if module_name not in module_to_packages:
+            continue
+
+        for pkg in module_to_packages[module_name]:
+            req = "{}=={}".format(pkg, importlib_metadata.version(pkg))
```

Comment:
We need to handle pyspark dev versions that aren't available on PyPI at some point as well by stripping away the `dev0` suffix, as is done in https://github.com/mlflow/mlflow/blob/bc86f1dfdb6444746470b7b12b6582b1390eb1aa/mlflow/spark.py#L83. 

Perhaps we should introduce a flag that controls whether or not known package dependency versions that we know are private should be swapped for publicly-available versions.

Reply:
Thanks for pointing this out. Agree on introducing such flag.